### PR TITLE
Modify 2-poles to be 3-poles with tau3 = 0, C=0

### PR DESCRIPTION
**Pull request by serfass**
- For now, keep fitting to be separate between 2-poles (5 parameters) and 3-poles (7 parameters) however in "dofit", immediately modify result after fit  to add C=0 and tau3=0 in parameter list and covariance
- modify fit result internal dictionary from 3 dictionaries self._1/2/3poleresult to single self._fit_results dictionary with key=1,2,3.  
- Modify fitresult() function argument to be poles=None, if "poles"=1,2, or 3 return fit results as before (no change of internal structure to preserve back compatibility), however if poles=None return all poles. 
- modify calc small signal parameters calculation to be same for 2/3 poles, now calculate also ssp light for 2 poles
- reorganize notebooks to have a single didv/resolution notebook (that was in demos/resolutions), modfiy to have model poles as variable 

Testing done:
calculate dPdI and uncertainities  and resolution for both 2-poles and 3-poles with notebook https://github.com/spice-herald/QETpy/blob/fitresult_container/demos/didv/didv_and_resolution_tutorial.ipynb
(number of poles is set in cell 13)

### CODE REVIEW

File: qetpy/core/didv/_didv.py [hunk 1]
```diff
@@ -1020,63 +1013,71 @@ def calc_bias_params_infinite_loop_gain(self, poles=3,
...
-        # replace results
-        if poles == 2:
-            self._2poleresult  = results
-        elif  poles == 3:
-            self._3poleresult = results
-                      
+
+            self._fit_results[model_poles] = results
+                                  
     def dofit_with_true_current(self, offset_dict,
```
Comment (rkromani) [inline qetpy/core/didv/_didv.py:1086]: is this deprecated in the current version of the code? Or should it still be used?

File: qetpy/core/didv/_didv.py [hunk 2]
```diff
@@ -911,107 +948,63 @@ def calc_smallsignal_params(self, ivsweep_results,
...
+            #     -> automatically make approximation that loop gain infinite 
 
-        # 2-poles
-        if calc_2poles:
-
+            results = copy.deepcopy(self._fit_results[model_poles])
+                            
             # check if infinite loop gain needs to be done
             set_infinite_loop_gain = inf_loop_gain_approx
             if inf_loop_gain_approx == 'auto':
```
Comment (rkromani) [inline qetpy/core/didv/_didv.py:974]: Where is this being set? I don't see it as a variable you can set when calling dofit

Comment (serfass) [reply qetpy/core/didv/_didv.py:974]: You mean the variable "model_poles"? It is in a loop:
for model_poles in model_list:
      ...


model_list  can be [2,3 ] or just [2] or [3] depending is dofit(2) and/or dofit(3) wa called before.  Given that the same functions are now used for 2 and 3, we don't need to have anything specific for 2/3 as before so we can loop a loop

File: qetpy/core/didv/_didv.py (hunk 1 above)
Comment (serfass) [reply qetpy/core/didv/_didv.py:1086]: Are you referring to dofit_with_true_current function. Yes, I think it should be deprecated to calc_smallsignal_parameters (it should be doing same thing but for both poles).

However, I think we should further cleanup by having the "true current" calculation to be be completely separated.  The "true current" does not even need to be done on dIdV data, we could use randoms.  The code could be separate from dIdV.  The input should a dictionary or pandas Series with I0, I0_err, etc.  I0 could be from IV sweep, true current, or fake muons, or even didv with infinite loop gain. It is a bit too black box at the moment. Separating steps would make things more understandable.  We can always add a funtion dofit_and_calcI0_and calc_ssp(...) if we want...

**Comment by serfass** [comment]
I fixed the testing script so it passes all tests now.

**Review by rkromani — APPROVED** [review]

**Pull request merged**